dapp_runner/descriptor/manifest.py

```diff
@@ -0,0 +1,101 @@
+"""Manifest verification support."""
+import json
+import logging
+import re
+from base64 import standard_b64decode
+from datetime import datetime
+from typing import Optional
+
+from dateutil.tz import UTC
+from OpenSSL import crypto
+
+from yapapi.payload.manifest import Manifest
+
+from .dapp import PAYLOAD_RUNTIME_VM_MANIFEST, DappDescriptor, PayloadDescriptor
+
+logger = logging.getLogger(__name__)
+
+
+def _get_manifest(params) -> Optional[Manifest]:
+    manifest_str = params.get("manifest")
+    if manifest_str:
+        manifest_dict = json.loads(standard_b64decode(manifest_str))
+        return Manifest.parse_obj(manifest_dict, by_alias=True)
+
+    return None
+
+
+def _get_manifest_cert(params) -> Optional[crypto.X509]:
+    cert_str = params.get("manifest_cert")
+    if cert_str:
+        return crypto.load_certificate(crypto.FILETYPE_PEM, standard_b64decode(cert_str))
+
+    return None
+
+
+def verify_manifest(payload_name: str, payload: PayloadDescriptor) -> None:
+    """Verify a single payload manifest, if present."""
+
+    manifest = _get_manifest(payload.params)
+    cert = _get_manifest_cert(payload.params)
+    sig = standard_b64decode(payload.params.get("manifest_sig", ""))
+    sig_algorithm = payload.params.get("manifest_sig_algorithm", "")
+
+    now = datetime.now(UTC)
+
+    if manifest:
```

Comment:
Can there be manifest payload without manifest?

Reply:
yes, the manifest can be generated on the fly